server/routes/incidentDetails/incidentDetailsPage.ts

```diff
@@ -0,0 +1,728 @@
+      originalReporterUsernameForPage = data.temporaryData?.originalReporterUsername
+    }
+    const pageData = await this.getPageDataOnGet(requestValues, originalReporterUsernameForPage, data, user)
+    renderData(res, pageData, null)
+
+    // Only delete session data when the page is successfully shown.
+    // This ensures the user can still see the data they entered in case of error.
+    deleteSessionData(req)
+  }
+
+  submit = async (req: Request, res: Response): Promise<void> => {
+    const postValues = extractValuesFromPost(req)
+    const postData = getPageActionFromPost(postValues)
+    const { user } = res.locals
+    debugData('POST', postValues)
+    debugData('POST action', postData)
+
+    if (postData !== PageRequestAction.STANDARD) {
+      const dataToSaveAfterRedirect = transformFormDataToStashedData(postValues)
+      let returnUrl = null
+      if (this.pageOptions.isEdit()) {
+        if (this.pageOptions.isPreviouslySubmitted()) {
+          returnUrl = `/incident-details/${postValues.prisonerNumber}/${postValues.draftId}/submitted/edit${
+            postValues.originalPageReferrerUrl ? `?referrer=${postValues.originalPageReferrerUrl}` : ''
+          }`
+        } else {
+          returnUrl = `/incident-details/${postValues.prisonerNumber}/${postValues.draftId}/edit`
+        }
+      } else {
+        returnUrl = `/incident-details/${postValues.prisonerNumber}`
+      }
+      stashDataOnSession(returnUrl, dataToSaveAfterRedirect, req)
+      if (postData === PageRequestAction.DELETE_PRISONER) {
+        return redirectToDeletePersonPage(res, postValues.incidentDetails.currentAssociatedPrisonerNumber)
+      }
+      if (postData === PageRequestAction.SEARCH_FOR_PRISONER) {
+        const searchValidationError = validatePrisonerSearch({
+          searchTerm: postValues.incidentDetails.currentAssociatedPrisonerNumber,
+          inputId: postValues.incidentDetails.currentIncidentRoleSelection,
+        })
+        if (searchValidationError) {
+          // Could stash to session and redirect here
+          const pageData = await this.getPageDataOnPost(postValues, user)
+          return renderData(res, pageData, searchValidationError)
+        }
+        return redirectToSearchForPersonPage(res, postValues.incidentDetails.currentAssociatedPrisonerNumber)
+      }
+    }
+
+    const validationError = validateForm({
+      incidentDate: postValues.incidentDetails?.incidentDate,
+      locationId: postValues.incidentDetails?.locationId,
+      incidentRole: postValues.incidentDetails?.currentIncidentRoleSelection,
+      associatedPrisonersNumber: postValues.incidentDetails?.currentAssociatedPrisonerNumber,
+    })
+    if (validationError) {
+      // Could stash to session and redirect here
+      const pageData = await this.getPageDataOnPost(postValues, user)
+      return renderData(res, pageData, validationError)
+    }
+
+    const incidentDetailsToSave = postValues.incidentDetails
+
+    try {
```

Comment:
Out of interest - why would we get an exception here? Should we just let it throw and get the sorry screen?

Reply:
I was preserving the existing functionality - but it does things like set up a specific URL for the Sorry page on error